### DISCUSSION TITLE
컴파일 오류들

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#76] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/62102830/198996383-d354321c-a8eb-4bea-80d0-0f0f5c5e5a92.png)
멤버 함수의 이름과 멤버 변수의 이름이 같을 수 없다는 사실을 깨달았다.

![image](https://user-images.githubusercontent.com/62102830/198997197-5403de2b-6ecc-4271-9b25-3fdf8874c7b3.png)
클래스 이름이 자료형으로 인식되지 않는 문제점.
#define 때문인것 같다. 해결 방법은 아직 찾지 못함.

**Comment by Woon-2** [comment, marked as answer]
보통 `class`용으로 `conditional compilation`문을 작성할 때 이용하는 `macro`의 이름은,   
`class`의 이름 앞에 `_`를 붙입니다.

```cpp
#ifndef _my_string
#define _my_string

class my_string { ... };

#endif
```

와 같은 느낌으로 말이죠.
